Add a post acknowledging all people involved in Spyder 4 (#52)
Quansight-Labs/quansight-labs-site: posts/2020/03/Spyder-4.md

<!--
.. title: Spyder 4 is here!
.. slug: spyder-4-is-here
.. date: 2020-03-05 14:00:00 UTC-05:00
.. author: Carlos Córdoba
.. tags: Labs, Spyder
.. category:
.. link:
.. description:
.. type: text
-->

> After more than three years in development, and comprising more than 5000 commits from 60 authors around the world, Spyder 4 finally saw the light in December 5th 2019! I decided to wait until now to make a formal announcement because quickly after the release we found several critical performance issues and some regressions with respect to Spyder 3, most of which are fixed now in version [4.1.0](https://github.com/spyder-ide/spyder/releases), released on <...>.

Reviewer: ```suggestion
After more than three years in development and more than 5000 commits from 60 authors around the world, Spyder 4 finally saw the light on December 5, 2019! I decided to wait until now to make a formal announcement because shortly after the initial release, we found several critical performance issues and some regressions with respect to Spyder 3, most of which are fixed now in [version 4.1.0](https://github.com/spyder-ide/spyder/releases), released on <...>.
```

"Saw the light" isn't particularly idiomatic for this context, but I'm not exactly sure what to replace it with that would fit your vision so I just left it.

<!-- TEASER_END -->

> This new release comes with a lengthy list of user-requested features aimed at providing users an enhanced development experience at the level of top general-purpose editors and IDEs, while strengthening Spyder's specialized focus on scientific programming in Python. The interested reader can take a look at some of them in [previous](https://labs.quansight.org/blog/2019/11/variable-explorer-improvements-in-Spyder-4/) [blog](https://labs.quansight.org/blog/2019/11/File-management-improvements-in-Spyder4/) [posts](https://labs.quansight.org/blog/2019/08/spyder-40-beta4-kite-integration-is-here/), and in detail in our [Changelog](https://github.com/spyder-ide/spyder/blob/master/CHANGELOG.md#version-400-2019-12-06). However, this post is not meant to describe these improvements but to acknowledge all people that contributed to make Spyder 4 possible.

Reviewer: ```suggestion
This new release comes with a lengthy list of user-requested features aimed at providing an enhanced development experience at the level of top general-purpose editors and IDEs, while strengthening Spyder's specialized focus on scientific programming in Python. The interested reader can take a look at some of them in [previous](https://labs.quansight.org/blog/2019/11/variable-explorer-improvements-in-Spyder-4/) [blog](https://labs.quansight.org/blog/2019/11/File-management-improvements-in-Spyder4/) [posts](https://labs.quansight.org/blog/2019/08/spyder-40-beta4-kite-integration-is-here/), and in detail in our [Changelog](https://github.com/spyder-ide/spyder/blob/master/CHANGELOG.md#version-400-2019-12-06). However, this post is not meant to describe those improvements, but to acknowledge all people that contributed to making Spyder 4 possible.
```

First sentence sounds a little familiar :)

> Spyder 4 is the first version developed with financial support from several companies as well as donations by the international user community. But, as a project, we couldn't have been able to reach the level of maturity needed to receive and handle that support, without the pivotal opportunities Travis Oliphant, former CEO of Anaconda and current one of Quansight, gave me to work in Scientific Python. Thanks to him, I became part of Anaconda, Inc. in 2015, I was able to hire a small small team to improve Spyder within Anaconda in late 2016, and I entered Quansight to work solely on Spyder in 2018. As with other projects in our ecosystem, such as [Bokeh](https://github.com/bokeh/bokeh), [Dask](https://github.com/dask/dask) and [Numba](https://github.com/numba/numba), Spyder benefited immensely from Travis' trust in the role these projects and ours could play in the future. He certainly believed in the vision their maintainers worked so hard to make a reality of, even if their beginnings were humble and their chances of survival uncertain. Therefore, my first big acknowledgment is to Travis: thanks for giving us a chance!

Reviewer: ```suggestion
Spyder 4 is the first version developed with financial support from multiple companies, as well as donations by the international user community. However, as a project, we couldn't have been able to reach the level of maturity needed to receive and handle that support without the pivotal opportunities Travis Oliphant, former CEO of Anaconda and current leader of Quansight, gave me to work in Scientific Python. Thanks to him, I became part of Anaconda Inc. in 2015; I was able to hire a small small team to improve Spyder within Anaconda in late 2016; and I was hired by Quansight to work solely on Spyder in 2018. As with other projects in our ecosystem, such as [Bokeh](https://github.com/bokeh/bokeh), [Dask](https://github.com/dask/dask) and [Numba](https://github.com/numba/numba), Spyder benefited immensely from Travis' trust in the role these efforts and ours could play in the future. He certainly believed in the vision their maintainers worked so hard to make a reality, even if their beginnings were humble and their chances of survival uncertain. Therefore, my first big acknowledgment is to Travis: thanks for giving us a chance!
```

I thought Travis himself was the original developer/maintainer of Numba?

This is _very_ well written.

> I also want to thank our community for its continued support. As I've witnessed during my years as Spyder's lead developer, many newbies and veterans alike keep choosing Spyder as a viable alternative to do scientific programming in Python. It's really you, the members of this fantastic community, which maintain Spyder relevant in a highly competed field by using and contributing back to it. I'm honored and humbled to have received such first-class additions to Spyder 4 from our volunteer contributors as our new Plots pane, thanks to [Jean-Sébastien Gosselin](https://github.com/jnsebgosselin); a complete re-architecturing of our IPython console, which allowed a lot of improvements to our debugger, thanks to [Quentin Peter](https://github.com/impact27); auto-saving functionality for our editor, thanks to [Jitse Niesen](https://github.com/jitseniesen); the initial implementation of the `runcell` command, thanks to [Brian Olsen](https://github.com/bcolsen); and significant improvements to our documentation, thanks to [CAM Gerlach](https://github.com/CAM-Gerlach).

Reviewer: ```suggestion
I also want to thank our community for its continued support. As I've witnessed during my years as Spyder's lead developer, many newbies and veterans alike keep choosing Spyder as their primary tool for scientific programming in Python. It's really you, the members of this fantastic community, which keep Spyder relevant in a highly competitive field by using and contributing back to it. I'm honored and humbled to have received such first-class additions to Spyder 4 from our volunteer contributors as our new Plots pane, thanks to [Jean-Sébastien Gosselin](https://github.com/jnsebgosselin); a complete re-archetecting of our IPython console, which enabled numerous improvements to our debugger, thanks to [Quentin Peter](https://github.com/impact27); auto-saving functionality for our editor, thanks to [Jitse Niesen](https://github.com/jitseniesen); the initial implementation of the `runcell` command, thanks to [Brian Olsen](https://github.com/bcolsen); and significant improvements to our documentation, thanks to [CAM Gerlach](https://github.com/CAM-Gerlach).
```

Technically speaking, Jean-Sébastien, Jitse, Brian and I were all on the core dev team during our primary time working on Spyder 4, not just community contributors, and the improvements to the docs I made were for Spyder 3, not 4. I don't feel its out of place for me (if I belong here at all) since I didn't contribute to Spyder 4 much at all, the vast majority of my work was for Spyder 3, the Spyder 3 docs, the Spyder 3 issue tracker, all that other stuff. But I would be concerned that mentioning them here could come across as putting them in a lesser, lower category compared to the other core devs mentioned below when, to my understanding, that wasn't the case at the time.

Author: > I don't feel its out of place for me (if I belong here at all) since I didn't contribute to Spyder 4 much at all, the vast majority of my work was for Spyder 3, the Spyder 3 docs, the Spyder 3 issue tracker, all that other stuff.

Ok, I'll leave here then.

> But I would be concerned that mentioning them here could come across as putting them in a lesser, lower category compared to the other core devs mentioned below when, to my understanding, that wasn't the case at the time.

I agree with this. I'll merge this paragraph with the one below about the Colombian team.

> Besides, when the future looked grim, after my team and I were let go from Anaconda at the end of 2017 (not by Travis decision), a lot of users came to our rescue by making donations through our [Open Collective page](https://opencollective.com/spyder). That, and a Numfocus development grant we received next year, filled us with confidence and allowed us to continue with Spyder's development in 2018, even after losing part of our team in the process.

Reviewer: ```suggestion
Furthermore, when the future looked grim, after my team and I were let go from Anaconda at the end of 2017 (not by Travis' decision), a lot of users came to our rescue by making donations through our [Open Collective page](https://opencollective.com/spyder). That, and a Numfocus development grant we received the next year, filled us with confidence and allowed us to continue with Spyder's development in 2018, even after losing part of our team in the process.
```

> Last year it was also a pleasant surprise to learn that several companies were interested in seeing Spyder prosper and thrive. Through Quansight Labs and its [Community Working Order](https://labs.quansight.org/blog/2019/05/community-driven-opensource-funded-development/) concept, we were able to sign contracts with two of them: [TDK-Micronas](https://www.micronas.tdk.com/en) and [Kite](https://kite.com/). Their support was critical to finish Spyder 4 because it allowed me to hire most of my old Anaconda team back, plus two new additions, to work full-time on the project. Therefore, I can't them enough for showing up just at the right time!

Reviewer: ```suggestion
Last year it was also a pleasant surprise to learn that several companies were interested in seeing Spyder prosper and thrive. Through Quansight Labs and its [Community Work Order](https://labs.quansight.org/blog/2019/05/community-driven-opensource-funded-development/) concept, we were able to sign contracts with two of them: [TDK-Micronas](https://www.micronas.tdk.com/en) and [Kite](https://kite.com/). Their support was critical to finish Spyder 4 because it allowed me to hire most of my old Anaconda team back, plus two new additions, to work full-time on the project. Therefore, I can't them enough for showing up just at the right time!
```

To be fair, I'm not saying you should change it but realistically, I highly doubt Kite gives a shit about Spyder itself, they just see at as a marketing platform to agglomerate more users, just like they did with the different open-source projects they bought out in the past (although, to be clear, their present behavior is vastly less objectionable than what it used to be, I don't think you would have stood for the kind of nonsense they used to pull). Assuming that funding went to substantively more than just integrating Kite, as I assume it did, I don't see it as a net negative for Spyder but at least between you and I, we can recognize it for what it is.

> And finally, even when I am the public face of Spyder, due to my presence in our [issues tracker](https://github.com/spyder-ide/spyder/issues) and [StackOverflow](https://stackoverflow.com/users/438386/carlos-cordoba), it's really the Spyder team that is in charge of implementing new features and fixing most bugs. So my last round of acknowledgments goes to them. I was fortunate enough to convince some of the most talented Colombian software developers to work for the project, and they did a hell of a job in this release!

Reviewer: ```suggestion
And finally, even when I am often seen as the public face of Spyder, due to my presence in our [issue tracker](https://github.com/spyder-ide/spyder/issues) and [Stack Overflow](https://stackoverflow.com/users/438386/carlos-cordoba), it's really the Spyder team that is in charge of implementing new features and fixing most bugs. So my last round of acknowledgments goes to them. I was fortunate enough to convince some of the most talented Colombian software developers to work for the project, and they did a hell of a job in this release!
```

In reference to the above, this directly implies that the various core devs are not part of the actual "Spyder team", which may cause some hard feelings and questioning of their value to the project. I know it would for me if I was.

Author: Agreed. Will do the change you suggested.

> For instance, [Gonzalo Peña-Castellanos](https://github.com/goanpeca/) helped us to greatly improve the usability experience of code completion and linting in the editor, to implement most enhancements to the Files pane, and to refactor and put in good shape our configuration system; [Edgar Margffoy](https://github.com/andfoy) single-handedly created a client to support the same [protocol](https://microsoft.github.io/language-server-protocol/) used by VSCode to provide completion and linting for lots of programming languages, added snippet completions and vastly improved code folding in the editor; [Daniel Althviz](https://github.com/dalthviz) developed the necessary infrastructure to install and use Kite smoothly within Spyder, and added the new object explorer viewer to the Variable Explorer; and our Junior developers, [Stephannie Jimenez](https://github.com/steff456) and [Juanita Gomez](https://github.com/juanis2112), although still learning their way in our complex source code, managed to do important contributions, such as improving the icons we use per file type in Files (Juanita), and allowing to run code in an external system terminal on macOS (Stephannie).

Reviewer: ```suggestion
[Gonzalo Peña-Castellanos](https://github.com/goanpeca/) helped us to greatly improve the user experience of code completion and linting in the editor, implemented most of the enhancements to the Files pane, and refactored and improved our configuration system; [Edgar Margffoy](https://github.com/andfoy) single-handedly created a client to support the same [protocol](https://microsoft.github.io/language-server-protocol/) used by VSCode to provide completion and linting for lots of programming languages, added snippet completions and vastly improved code folding in the editor; [Daniel Althviz](https://github.com/dalthviz) developed the necessary infrastructure to install and use Kite smoothly within Spyder, and added the new object viewer to the Variable Explorer; and our junior developers, [Stephannie Jimenez](https://github.com/steff456) and [Juanita Gomez](https://github.com/juanis2112), although still finding their way around our complex codebase, managed to make important contributions, such as improving the icons we use per file type in Files (Juanita), and allowing users to run code in an external system terminal on macOS (Stephannie).
```

> I hope you enjoy the results of this massive effort! And happy Spydering!!

Reviewer: ```suggestion
I hope you all enjoy the results of this massive effort! And happy Spydering!!!
```

Pro tip: In English its convention to either use one exclamation point, or (very sparingly) three. Two just looks like a mistake.

Author: Thanks for the suggestion!

<!-- TEASER_END -->

This new release comes with a lengthy list of user-requested features aimed at providing an enhanced development experience at the level of top general-purpose editors and IDEs, while strengthening Spyder's specialized focus on scientific programming in Python.
The interested reader can take a look at some of them in [previous](https://labs.quansight.org/blog/2019/11/variable-explorer-improvements-in-Spyder-4/) [blog](https://labs.quansight.org/blog/2019/11/File-management-improvements-in-Spyder4/) [posts](https://labs.quansight.org/blog/2019/08/spyder-40-beta4-kite-integration-is-here/), and in detail in our [Changelog](https://github.com/spyder-ide/spyder/blob/master/CHANGELOG.md#version-400-2019-12-06).
However, this post is not meant to describe those improvements, but to acknowledge all people that contributed to making Spyder 4 possible.

> Spyder 4 is the first version developed with financial support from multiple companies, as well as donations by the international user community.

Reviewer: ```suggestion
Spyder 4 is the first version developed with financial support from multiple companies and organizations, as well as donations by the international user community.
```

Emphasize its not all corporate donations (e.g. NumFOCUS, etc)

Thanks to him, I became part of Anaconda Inc. in 2015; I was able to hire a small small team to improve Spyder within Anaconda in late 2016; and I was hired by Quansight to work solely on Spyder in 2018.
As with other projects in our ecosystem, such as [Bokeh](https://github.com/bokeh/bokeh), [Dask](https://github.com/dask/dask) and [Numba](https://github.com/numba/numba), Spyder benefited immensely from Travis' trust in the role these efforts and ours could play in the future.
He certainly believed in the vision their maintainers worked so hard to make a reality, even if their beginnings were humble and their chances of survival uncertain.
Therefore, my first big acknowledgment is to Travis: thanks for giving us a chance!

I also want to thank our community for its continued support.
As I've witnessed during my years as Spyder's lead developer, many newbies and veterans alike keep choosing Spyder as their primary tool for scientific programming in Python.
It's really you, the members of this fantastic community, which keep Spyder relevant in a highly competitive field by using and contributing back to it.

> I'm also honored and humbled to have received such first-class additions to Spyder 4 from our volunteer contributors as significant improvements to our documentation, thanks to [CAM Gerlach](https://github.com/CAM-Gerlach).

Reviewer: It seems kinda strange to have just my name singled out up here, and everybody else down there in one big blob; Jitse, Brian, Quentin, and Jean-Sebastian were volunteer contributors and dev team members just like I was, and the syntax sounds weird only listing one person.

Author: I agree. I'll do this change.

Last year it was also a pleasant surprise to learn that several companies were interested in seeing Spyder prosper and thrive.
Through Quansight Labs and its [Community Work Order](https://labs.quansight.org/blog/2019/05/community-driven-opensource-funded-development/) concept, we were able to sign contracts with two of them: [TDK-Micronas](https://www.micronas.tdk.com/en) and [Kite](https://kite.com/).
Their support was critical to finish Spyder 4 because it allowed me to hire most of my old Anaconda team back, plus two new additions, to work full-time on the project.
Therefore, I can't thank them enough for showing up just at the right time!

And finally, even when I am often seen as the public face of Spyder, due to my presence in our [issue tracker](https://github.com/spyder-ide/spyder/issues) and [Stack Overflow](https://stackoverflow.com/users/438386/carlos-cordoba), it's really the Spyder team that is in charge of implementing new features and fixing most bugs.
So my last round of acknowledgments goes to them.

> I was fortunate enough to convince some of the most talented Colombian software developers to work for the project, and to attract the interest of several other equally accomplished developers from around the world. All of them did a hell of a job in this release!

Reviewer: ```suggestion
I was fortunate enough to convince some of the most talented Colombian software developers to work for the project, and to attract the interest of several other equally accomplished developers from around the world.
All of them did a hell of a job in this release!
```

Missing sentence break.

> [Jean-Sébastien Gosselin](https://github.com/jnsebgosselin) contributed our new Plots pane in its entirety; [Quentin Peter](https://github.com/impact27) did a complete re-architecting of our IPython console, which enabled numerous improvements to our debugger; [Jitse Niesen](https://github.com/jitseniesen) added auto-saving functionality to our editor; [Brian Olsen](https://github.com/bcolsen) contributed the initial implementation of the `runcell` command; [Gonzalo Peña-Castellanos](https://github.com/goanpeca/) helped us to greatly improve the user experience of code completion and linting in the editor, implemented most of the enhancements to the Files pane, and refactored and improved our configuration system; [Edgar Margffoy](https://github.com/andfoy) single-handedly created a client to support the same [protocol](https://microsoft.github.io/language-server-protocol/) used by VSCode to provide completion and linting for lots of programming languages, added code snippet completions and vastly improved code folding in the editor; [Daniel Althviz](https://github.com/dalthviz) developed the necessary infrastructure to install and use Kite smoothly within Spyder, and added the new object viewer to the Variable Explorer; and our junior developers, [Stephannie Jimenez](https://github.com/steff456) and [Juanita Gomez](https://github.com/juanis2112), although still finding their way around our complex codebase, managed to make important contributions, such as improving the icons we use per file type in Files (Juanita), and allowing users to run code in an external system terminal on macOS (Stephannie).

Reviewer: A really really really long sentence, I broke it up a bit but didn't change anything otherwise.

```suggestion
[Jean-Sébastien Gosselin](https://github.com/jnsebgosselin) contributed our new Plots pane in its entirety; [Quentin Peter](https://github.com/impact27) did a complete re-architecting of our IPython console, which enabled numerous improvements to our debugger; [Jitse Niesen](https://github.com/jitseniesen) added auto-saving functionality to our editor; and [Brian Olsen](https://github.com/bcolsen) contributed the initial implementation of the `runcell` command.
[Gonzalo Peña-Castellanos](https://github.com/goanpeca/) helped us to greatly improve the user experience of code completion and linting in the editor, implemented most of the enhancements to the Files pane, and refactored and improved our configuration system; [Edgar Margffoy](https://github.com/andfoy) single-handedly created a client to support the same [protocol](https://microsoft.github.io/language-server-protocol/) used by VSCode to provide completion and linting for lots of programming languages, added code snippet completions and vastly improved code folding in the editor; [Daniel Althviz](https://github.com/dalthviz) developed the necessary infrastructure to install and use Kite smoothly within Spyder, and added the new object viewer to the Variable Explorer.
Finally, our junior developers, [Stephannie Jimenez](https://github.com/steff456) and [Juanita Gomez](https://github.com/juanis2112), although still finding their way around our complex codebase, managed to make important contributions, such as improving the icons we use per file type in Files (Juanita), and allowing users to run code in an external system terminal on macOS (Stephannie).
```

> I hope you all enjoy the results of this massive effort! And happy Spydering!!!

Reviewer: Sentence breaks

```suggestion
I hope you all enjoy the results of this massive effort!
And happy Spydering!!!
```
